spec/requests/custom_actions_spec.rb

```diff
@@ -220,4 +220,38 @@ def expect_result_to_have_custom_actions_hash
       expect(response.parsed_body).to include(expected)
     end
   end
+
+  def define_custom_button1(resource)
+    dialog1          = FactoryGirl.create(:dialog, :label => "dialog1")
+    resource_action1 = FactoryGirl.create(:resource_action, :dialog_id => dialog1.id)
+    FactoryGirl.create(:custom_button, :name => "button1", :applies_to => resource, :userid => @user.userid, :resource_action => resource_action1)
+  end
+
+  describe "Generic Objects" do
+    before do
+      @object_definition = FactoryGirl.create(:generic_object_definition, :name => 'object def')
+      @resource = FactoryGirl.create(:generic_object, :generic_object_definition => @object_definition)
+      define_custom_button1(@object_definition)
+    end
+
+    it "queries return custom actions defined" do
+      api_basic_authorize(action_identifier(:generic_objects, :read, :resource_actions, :get))
+
+      get api_generic_object_url(nil, @resource)
+
+      expect(response.parsed_body).to include(
+        "id"      => @resource.id.to_s,
+        "href"    => api_generic_object_url(nil, @resource),
+        "actions" => a_collection_including(a_hash_including("name" => "button1"))
+      )
+    end
+
+    it "accept custom actions" do
+      api_basic_authorize
+
+      post api_generic_object_url(nil, @resource), :params => gen_request(:button1, "key1" => "value1")
+
+      expect_single_action_result(:success => true, :message => /.*/, :href => api_generic_object_url(nil, @resource))
+    end
+  end
```

Comment:
can you add a test that actions returns both custom actions from custom buttons as well as the actions from the generic object definition methods ?